src/shogun/lib/parameter_observers/ObservedValueTemplated.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Giovanni De Toni
+ *
+ */
+
+#ifndef SHOGUN_OBSERVEDVALUETEMPLATED_H
+#define SHOGUN_OBSERVEDVALUETEMPLATED_H
+
+#include <shogun/base/Parameter.h>
+#include <shogun/lib/parameter_observers/ObservedValue.h>
+
+namespace shogun
+{
+
+	/**
+	 * Templated specialisation of ObservedValue that stores the actual data.
+	 * @tparam T the type of the observed value
+	 */
+	template <class T>
+	class ObservedValueTemplated : public ObservedValue
+	{
+
+	public:
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name the observed value's name
+		 * @param value the observed value
+		 */
+		ObservedValueTemplated(int64_t step, std::string name, T value)
+		    : ObservedValue(step, name), m_observed_value(value)
+		{
+			this->watch_param(
+			    "value", &m_observed_value,
+			    AnyParameterProperties(
+			        "Value of the observation", ParameterProperties::NONE));
```

Comment:
Actually, I think the ctor should also accept a description of the observation, rather than just putting it "Value of the observation". So when observations are made (especially for the custom observations like the CV one), users can actually get a one-liner what this is about? What are your thoughts on that?

Reply:
Ideally, the `ObservedValue` has already a name which should refer to its content and it should be enough. The `m_observed_value` instead is the actual observed value, so its description specifies only that if we do `get("value")` we will get the observed data.

Reply:
Not sure I agree....think a single name "weights"...what weights would those be? Better would be "weights" and "MKL kernel weights of the corresponding x-validation fold". No?

Reply:
mmh yeah, I guess it can make it more understandable. I'll update the constructor to take also a description.

Reply:
Updated.